File: DataStructures/StacksQueues/implementing_stack_array.py
class Stack:
    def __init__(self):
        self.array = []
    
    def peek(self):
        # See the last item
        if not self.array:
            return None
        return self.array[-1]
    
    def push(self, value):
        # Add the item to the top of the stack
        self.array.append(value)
        return self.array
    
    def pop(self):
        # Remove the last item
        return self.array.pop()
    
    def is_empty(self):
        # See whether a stack is empty
        return not bool(self.array)


if __name__ == "__main__":
    my_stack = Stack()
    print(f"Stack is empty: {my_stack.is_empty()}")
    my_stack.push(2)
    print(f"After push(2) stack is empty: {my_stack.is_empty()}")
    print(f"The last item in stack: {my_stack.peek()}")
    my_stack.push(True)
    print(f"The last item after push(True): {my_stack.peek()}")
    print(f"Popping the last item: {my_stack.pop()}")
    print(f"The last item in stack: {my_stack.peek()}")

File: DataStructures/StacksQueues/implementing_stack_array.js
class Stack {
    constructor() {
        this.array = [];
    }

    peek() {
        // see the last item
        return this.array[this.array.length-1];
    }

    push(value) {
        // add the item to the top of the stack
        this.array.push(value);
        return this.array;
    }

    pop() {
        // remove the last item
        return this.array.pop();
    }

    isEmpty() {
        // see whether a stack is empty
        return !Boolean(this.array.length);
    }
}

if (require.main === module) {
    let myStack = new Stack();
    console.log("Stack is empty:", myStack.isEmpty());
    myStack.push(2);
    console.log("After push(2) stack is empty:", myStack.isEmpty());
    console.log("The last item in stack:", myStack.peek());
    myStack.push(true);
    console.log("The last item after push(true):", myStack.peek());
    console.log("Popping the last item:", myStack.pop());
    console.log("The last item in stack:", myStack.peek());
}
